sdk/identity/Azure.Identity/src/AzureCliCredentialClient.cs

```diff
@@ -0,0 +1,201 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Text.Json;
+using System.IO;
+using System.Collections.Generic;
+using Azure.Core;
+using System.Globalization;
+using System;
+using System.Text.RegularExpressions;
+using Azure.Core.Pipeline;
+
+namespace Azure.Identity
+{
+    internal class AzureCliCredentialClient
+    {
+        private const string AzureCLINotInstalled = "Azure CLI not installed";
+        private const string AzNotLogIn = "Please run 'az login' to set up account";
+        private const string WinAzureCLIError = "'az' is not recognized";
+        private const string AzureCliTimeoutError = "Azure CLI authentication timed out.";
+        private const string AzureCliFailedError = "Azure CLI authentication failed due to an unknown error.";
+        private const int CliProcessTImeoutMs = 10000;
+        private readonly string _workingDir;
+        private readonly string _path;
+
+        // The default install paths are used to find Azure CLI if no path is specified. This is to prevent executing out of the current working directory.
+        private static readonly string DefaultPathWindows = $"{EnvironmentVariables.ProgramFilesX86}\\Microsoft SDKs\\Azure\\CLI2\\wbin;{EnvironmentVariables.ProgramFiles}\\Microsoft SDKs\\Azure\\CLI2\\wbin";
+        private const string DefaultPath = "/usr/bin:/usr/local/bin";
+        private const string DefaultWorkingDir = "/bin/";
+        private static readonly string DefaultWorkingDirWindows = Environment.GetFolderPath(Environment.SpecialFolder.System);
+
+        private static readonly Regex AzNotFoundPattern = new Regex("az:(.*)not found");
+
+        public AzureCliCredentialClient()
+        {
+            _path = EnvironmentVariables.Path;
+
+            if (string.IsNullOrEmpty(_path))
+            {
+                _path = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? DefaultPathWindows : DefaultPath;
```

Comment:
Is `DefaultPath` identical for Mac and Linux?

Reply:
Yes for now we default to /bin/ on all posix based systems.

Reply:
this _path variable, is no longer used, since we use default paths below.